openstf/pipeline/optimize_hyperparameters.py

```diff
@@ -22,13 +25,59 @@
 TIMEOUT: int = 200  # Stop study after the given number of second(s).
 TRAIN_HORIZONS: List[float] = [0.25, 24.0]
 
+# The dictonairy can be interpret the following way
+# Tuples; E.g. name:( (min, max), logarithmic); for intigers and floats
+#           or name:((option1, option2), logarithmic); picks option 2 for lgb and for all others option 1
+# List; E.g. name:[option1, option2, ...]; for categorical
+default_paramspace: dict = {
+    # General parameters
+    "learning_rate": ((0.01, 0.2), True),
+    "alpha": ((1e-8, 1.0), True),
+    "lambda": ((1e-8, 1.0), True),
+    "subsample": ((0.5, 0.99), False),
+    "min_child_weight": ((1, 6), False),
+    "max_depth": ((3, 10), False),
+    "colsample_bytree": ((0.5, 1.0), False),
+    "max_delta_step": ((1, 10), False),
+}
+# Important parameters, model specific
+# XGB specific
+xgb_paramspace: dict = {
```

Comment:
I think the model specific stuff should probably not be part of the pipeline? Think about responsibilities? The pipeline should probably be only responsible for conduction the optimization and not have any knowledge about different model implementations. I guess the model specific stuff should either be part of a model specific optimizer or if its really small perhaps part of the model itself.